src/functionalTest/groovy/com/autonomousapps/jvm/IncludedBuildSpec.groovy

```diff
@@ -19,15 +20,34 @@ final class IncludedBuildSpec extends AbstractJvmSpec {
     gradleProject = project.gradleProject
 
     when: 'build does not fail'
-    build(gradleVersion, gradleProject.rootDir, 'buildHealth')
+    build(gradleVersion, gradleProject.rootDir, ':buildHealth')
```

Comment:
why did you add the `:` prefix here?

one of the constraints of the way this plugin is currently architected is that we generally need to invoke the task as `buildHealth` to force gradle to configure all of the subprojects, which is how results get aggregated into the `buildHealth` report.

Reply:
Background: When you call a task in an included build, the fuzzy matching (without `:`) is not supported. Although I never saw anything official about that, it was discussed that it is generally better to be precise when invoking Gradle (i.e. it is idiomatic to always use `:` notation). And if you want an entry point that calls a number of tasks at once, you should create a _lifecycle task_ with multiple _dependsOn_. (This is rather cumbersome right now, but that's how things currently are if you work with a larger included build setup.)

In these tests I wanted to be consistent with calling `:buildHealth`, and for the included build there is no other way to call it:

- `:buildHealth` build health of the root build (could also call `buildHealth`
- `:second:buildHealth` build health of the _second_ build (no other way to call it)

Generally, I think doing `:buildHealth` is fine as there is only one `buildHealth` task per build. Gradle always configures all the subprojects of a build independent of which task you call. Such considerations might only be important for the old _configure-on-demand_ flag, which is half broken anyway in my opinion (and will be removed I think soonish).

If you prefer to change this back to `buildHealth` in the tests wherever possible, I can do that. I have no strong opinion on that.

Reply:
> Gradle always configures all the subprojects of a build independent of which task you call.

This was not my experience writing these tests. At one point I tried to be more idiomatic and added the `:` prefix, and the tests started failing. Maybe there was a bug somewhere. And I am also certain that I ran into it at work, where we use do CoD.

If your tests are passing (they are), and the assertions make sense to you, then you can leave it as it is.